homeassistant/components/solaredge/sensor.py

```diff
@@ -425,22 +425,21 @@ def update(self):
         self.data = {}
         self.attributes = {}
         self.unit = energy_details["unit"]
-        meters = energy_details["meters"]
-
-        for entity in meters:
-            for key, data in entity.items():
-                if key == "type" and data in [
-                    "Production",
-                    "SelfConsumption",
-                    "FeedIn",
-                    "Purchased",
-                    "Consumption",
-                ]:
-                    energy_type = data
-                if key == "values":
-                    for row in data:
-                        self.data[energy_type] = row["value"]
-                        self.attributes[energy_type] = {"date": row["date"]}
+
+        for meter in energy_details["meters"]:
+            if "type" not in meter or "values" not in meter:
+                continue
+            if meter["type"] not in [
+                "Production",
+                "SelfConsumption",
+                "FeedIn",
+                "Purchased",
+                "Consumption",
+            ]:
+                continue
+            if len(meter["values"][0].keys()) == 2:
```

Comment:
This seems different from the previous version in two ways:
1. It uses the first row rather than the last
2. It will fail in case there are no rows

Are those intentional?

Reply:
1- there should only be one row (according to the spec)
2- this is intentional, as then the data is corrupt